src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -146,7 +146,10 @@ internal void SetProviderManager(IProviderManager manager)
 
         internal void AddProperty(string key, string val)
         {
-            properties.Add(key, val);
+            if (!properties.ContainsKey(key))
```

Comment:
This will silently fail to add the property if one already exists.  Maybe return bool: true == added, false == failed?